### PR TITLE
Cached invalidate annotation

### PR DESCRIPTION
**Pull request by zarinfam**
Add CachedInvalidate annotation and CachedInvalidateAction for Invalidate cache after trigger action.
https://github.com/playframework/playframework/issues/4019

### CODE REVIEW

File: framework/src/play-cache/src/main/java/play/cache/CachedInvalidateAction.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+/**
+ * Invalidate cache after trigger action.
+ */
+public class CachedInvalidateAction extends Action<CachedInvalidate> {
+
+    public F.Promise<Result> call(Context ctx) {
+        try {
+            final String[] keys = configuration.keys();
+            F.Promise<Result> promise = delegate.call(ctx);
+            for(String key: keys){
```
Comment (benmccann) [inline framework/src/play-cache/src/main/java/play/cache/CachedInvalidateAction.java]: need a space before the open brace

**Comment by jroper** [comment]
Could you describe the use cases a little more?  If we're going to include this, then we need tests as well as documentation in the Play manual about it.

**Comment by zarinfam** [comment]
See this scenario:
I cache the action result using Cached annotation:

    @Transactional(readOnly = true)
    @Cached(key = CACHE_HTTP_ALL_USERS)
    public Result getAllUsers() {
        return ok(Json.toJson(generalDao.getAll(User.class)));
    }

now i want to invalidate this cached result when the createUser action call:

    @Transactional
    @BodyParser.Of(BodyParser.Json.class)
    @CachedInvalidate(keys = CACHE_HTTP_ALL_USERS)
    public Result createUser() {
        generalDao.persist(Json.fromJson(request().body().asJson(), User.class));
        return Results.ok();
    }

or some situation like this. this is very handy and useful annotation for me.

**Comment by zarinfam** [comment]
I can help to document this feature but Play Framework tests is in Scala and i do not know much Scala.

**Comment by dotta** [comment]
@jroper What's your take on this PR?

**Comment by jroper** [comment]
The use case makes sense, but we really do need tests for it.  We should probably also have an equivalent API in Scala, and it should also be documented.

**Comment by zarinfam** [comment]
If you agree with this i will write test, document and Scala API for it in future.

**Comment by mkurz** [comment]
Ping @zarinfam :smile:

**Comment by zarinfam** [comment]
@mkurz  I am so busy, i am looking for some free time to do this task well.

**Comment by wsargent** [comment]
Closing due to inactivity

**Pull request closed without merging**